kyaml/setters2/list.go

```diff
@@ -181,12 +185,13 @@ func (l *List) listSubst(object *yaml.RNode) error {
 }
 
 // count returns the number of fields set by the setter with name
+// this excludes all the subpackages with openAPI file in them
 // set filter is leveraged for this but the resources are not written
 // back to files as only LocalPackageReader is invoked and not writer
 func (l *List) count(path, name string) (int, error) {
 	s := &Set{Name: name}
 	err := kio.Pipeline{
-		Inputs:  []kio.Reader{&kio.LocalPackageReader{PackagePath: path}},
+		Inputs:  []kio.Reader{&kio.LocalPackageReader{PackagePath: path, PackageFileName: l.OpenAPIFileName}},
```

Comment:
I saw the `PackageFileName` field in the LocalPackageReader when I was working on https://github.com/kubernetes-sigs/kustomize/pull/2898. Is there anytime when we don't want this set? It seems like we will be setting this field to the same value every time the LocalPackageReader is used.

Reply:
We should set this for each call as there could be nested packages within it which should be excluded for the count.

Reply:
There is a separate property on the LocalPackageReader that defines how nested packages should be treated. I think that would be the right way to decide how to handle nested packages. We can look at this when we do the other cfg commands.